### PR TITLE
support rolling-updates for cell components

### PR DESCRIPTION
**Pull request by isurulucky**

### CODE REVIEW

File: components/cli/pkg/commands/run.go [hunk 1]
```diff
@@ -1042,6 +995,47 @@ func getYamlFiles(path string) ([]string, error) {
 	return files, nil
 }
 
+// extractImage extracts the image into a temporary directory and returns the path.
+// Cleaning the path after finishing your work is your responsibility.
+func ExtractImage(cellImage *util.CellImage, spinner *util.Spinner) (string, error) {
```
Comment (MadhukaHarith92) [inline components/cli/pkg/commands/run.go]: Suggestion: CellImage struct can probably be moved to a different package, say `cell` package.

Comment (isurulucky) [reply components/cli/pkg/commands/run.go]: Noted, shall we move cellImage struct to image package?

Comment (isurulucky) [reply components/cli/pkg/commands/run.go]: This change introduces a circular dependency. Lets complete this as a part of the CLI refactoring effort.

File: components/cli/pkg/util/structs.go [hunk 2]
```diff
@@ -57,17 +59,37 @@ type CellAnnotations struct {
...
 type ComponentTemplate struct {
 	Metadata ComponentTemplateMetadata `json:"metadata"`
+	Spec     ComponentTemplateSpec     `json:"spec"`
 }
 
 type ComponentTemplateMetadata struct {
 	Name string `json:"name"`
 }
 
+type ComponentTemplateSpec struct {
```
[the remark this answers is not in the document]
Comment (isurulucky) [reply components/cli/pkg/util/structs.go]: moved.

File: components/cli/pkg/util/structs.go [hunk 3]
```diff
@@ -38,6 +38,8 @@ type CellList struct {
 }
 
 type Cell struct {
```
[the remark this answers is not in the document]
Comment (isurulucky) [reply components/cli/pkg/util/structs.go]: done.

File: components/cli/pkg/commands/run.go (hunk 1 above)
Comment (MadhukaHarith92) [reply components/cli/pkg/commands/run.go]: Ack.

**Pull request merged**